src/client-configuration.ts

```diff
@@ -111,6 +112,64 @@ export interface ClientConfiguration {
   typecheck?: boolean;
 }
 
+/**
+ * Options for queries. Each query can be made with different options.
+ */
+export interface QueryOptions {
```

Comment:
THis makes more sense in the `wire-protocol` file IMO as they effect the behavior of a query by transmitting things over the wire.

Reply:
Moved it back. Putting it in the "client configuration" file may be a bit of a misnomer. I keep feeling the desire to define the `ClientConfiguration` as an extension of `QueryOptions` since the fields and the types are the same (now they are the same since they are all optional), except... I just added `arguments` field which is now only in `QueryOptions`. Anyways, it still feels like a misfit in `wire-protocol` to me, but it's not something to argue about 🙂

Reply:
ok we can probably rename the files and put everything in one place in another PR. just thought it woudl be easier if the query is along side its options